api/managementpb/agent.proto

```diff
@@ -0,0 +1,114 @@
+  // Custom user-assigned labels.
+  map<string, string> custom_labels = 8;
+  // Desired Agent status: enabled (false) or disabled (true).
+  bool disabled = 9;
+  // List of disabled collector names.
+  repeated string disabled_collectors = 10;
+  // Listen port for scraping metrics.
+  uint32 listen_port = 11;
+  // Log level for exporter.
+  string log_level = 12;
+  // Limit query length in QAN.
+  int32 max_query_length = 13;
+  // Limit query log size in QAN.
+  int64 max_query_log_size = 14;
+  // Path under which metrics are exposed, used to generate URI.
+  string metrics_path = 15;
+  // Scheme to generate URI to exporter metrics endpoints.
+  string metrics_scheme = 16;
+  // TLS options for connecting to MongoDB.
+  string mongo_db_options = 17;
+  // Options for connecting to MySQL.
+  string mysql_options = 18;
+  // Node identifier.
+  string node_id = 19;
+  // Password for connecting the agent to the database.
+  string password = 20;
+  // The pmm-agent identifier which runs this instance.
+  string pmm_agent_id = 21;
+  // Options for connecting to PostgreSQL.
+  string postgresql_options = 22;
+  // Path to exec process.
+  string process_exec_path = 23;
+  // True if exporter uses push metrics mode.
+  bool push_metrics = 24;
+  // True if query examples are disabled.
+  bool query_examples_disabled = 25;
+  // True if RDS basic metrics are disdabled.
+  bool rds_basic_metrics_disabled = 26;
+  // True if RDS enhanced metrics are disdabled.
+  bool rds_enhanced_metrics_disabled = 27;
+  // Node identifier where this instance runs.
+  string runs_on_node_id = 28;
+  // Service identifier.
+  string service_id = 29;
+  // Actual Agent status.
+  string status = 30;
+  // Last known table count.
+  int32 table_count = 31;
+  // Tablestats group collectors are disabled if there are more than that number of tables.
+  // 0 means tablestats group collectors are always enabled (no limit).
+  // Negative value means tablestats group collectors are always disabled.
+  int32 table_count_tablestats_group_limit = 32;
+  // Use TLS for database connections.
+  bool tls = 33;
+  // Skip TLS certificate and hostname validation.
+  bool tls_skip_verify = 34;
+  // HTTP basic auth username for collecting metrics.
+  string username = 35;
+  // Last update timestamp.
+  int64 updated_at = 36;
+  // Agent version.
+  string version = 37;
+  // True if Agent is running and connected to pmm-managed.
+  bool is_connected = 38;
+}
+
+message ListAgentRequest {
+  // Return only Agents that provide insights for that ServiceID.
+  string service_id = 1;
+}
+
+message ListAgentResponse {
+  // List of Agent.
+  repeated GenericAgent agents = 1;
+}
+
+// Agent service provides public methods for managing and querying Agents.
+service Agent {
+  // ListAgents returns a list of Agents filtered by type.
+  rpc ListAgents(ListAgentRequest) returns (ListAgentResponse) {
+    option (google.api.http) = {
+      post: "/v1/management/Agent/List"
```

Comment:
That's weird, in `inventory` we are using plural naming like `Agents/List`, but in management it's singular

Reply:
Yes, and the `Service/{Remove,List}` are also singular. What shall we do?

Reply:
Let's stay consistent within the package at least and use singular.

Reply:
OK, I'll leave it as is then.